homework/src/index.js

```diff
@@ -31,17 +33,124 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('p', header, {html: 'HYF Repositories', class: 'header-lable'});
+    const select = createAndAppend('select', header, { id: 'selectMenu' });
+    // use promise object
+    const promise = fetchJSON(url);
+    promise.then(data => {
+      data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {ignorePunctuation: true}));
+      data.forEach((repository, index) => {
+        createAndAppend('option', select, {html: repository.name, value: index});
+      });
+      
+      const container = createAndAppend('div', root, { id: 'container'});
+      contentLeft(container, data, select);
+      contentRight(container, data, select);
+    })
+    .catch(error => {
+      createAndAppend('div', root, { html: error.message, class: 'alert-error' });
+    });
+  }
+
+
+
+
+
+  // function who creates a left content section
+  function contentLeft(container, data, select) {
+    const leftSection = createAndAppend('div', container, { id: 'leftSection', class: 'whiteframe' });
+    showTableRepo(leftSection, data[0]);
+
+    select.addEventListener('change', (event) => {
+      leftSection.innerHTML = "";
+      const theEventIndex = event.target.value;
+      showTableRepo(leftSection, data[theEventIndex]);
+    });
+  }
+
+  
+
+  // drow the table
+function showTableRepo(leftSection, repository) {
+  // create table of the contents
+  const lTable = createAndAppend('table', leftSection);
+  // line one
+  const tr1 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr1, {html: 'Repository :', class: 'lable'});
+  const td1 = createAndAppend('td', tr1);
+  const nameLink = '<a href="' + repository.html_url + '" target="_blank">' + repository.name + '</a>';
+  createAndAppend('p', td1, {html: nameLink });
+  // line two
+  const tr2 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr2, {html: 'Description :', class: 'lable'});
+  const td2 = createAndAppend('td', tr2);
+  createAndAppend('p', td2, {html: repository.description});
+  // line three
+  const tr3 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr3, {html: 'Forks :', class: 'lable'});
+  const td3 = createAndAppend('td', tr3);
+  createAndAppend('p', td3, {html: repository.forks});
+  // line four
+  const tr4 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr4, {html: 'Updated :', class: 'lable'});
+  const td4 = createAndAppend('td', tr4);
+  const defaultElDate = repository.updated_at;
```

Comment:
This is a comment from week 1:

Since you are repeating code here (create a `tr` with two `td`s as children, you could create a separate function to avoid repetition (the DRY principle: Don't Repeat Yourself), for instance:

```js
  function (table, label, value) {
    const tr = createAndAppend('tr', table);
    createAndAppend('td', tr, { html: `${label} :`, class: 'label' });
    createAndAppend('td', tr, { html: value });
  }
```